### ISSUE TITLE
recurse3() test case is failing when compiled

### ISSUE THREAD

**Issue by bobbimanners** [bobbimanners/EightBall#9]
```
sub recurse3(word x)
  if x==0
    return 1;
  else
    return x*recurse2(x-1)
  endif
endsub

```
This works okay in the interpreter but the compiled code does not run.  Swapping the order of the arguments to the multiplication resolves the issue!

**Comment by bobbimanners** [comment]
It seems the problem here is that the parser muddles up the outer (x * function_call) and the inner (x-1) expressions somehow.  The code emitted seems to try to calculate x*x-1.  This is probably a bug in the function call logic (which is a hack anyhow).

**Comment by bobbimanners** [comment]
Fixed in v0.62.
The problem was that the compiler was omitting to push SENTINEL to the operator stack before evaluating the function's operand, and pop it afterwards.  As a result the parsing was getting confused.
